### PR TITLE
[Neo Plugin Store] Unit test

### PR DESCRIPTION
**Pull request by Jim8y**
# Description

This is for testing the snapshot of stores

Fixes # (issue)

## Type of change

- [ ] Optimization (the change is only an optimization)
- [ ] Style (the change is only a code style for better maintenance or standard purpose)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [ ] TestLevelDbSnapshot
- [ ] TestLevelDbMultiSnapshot
- [ ] TestRocksDbSnapshot
- [ ] TestRocksDbMultiSnapshot

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by cschuchardt88 — APPROVED** [review]

File: tests/Neo.Plugins.Storage.Tests/StoreTest.cs [hunk 1]
```diff
@@ -114,18 +128,70 @@ public void TestLevelDbMultiSnapshot()
...
+            var snapshot2 = store.GetSnapshot();
+
+            var testKey = new byte[] { 0x01, 0x02, 0x03 };
+            var testValue = new byte[] { 0x04, 0x05, 0x06 };
+
+            snapshot.Put(testKey, testValue);
+            snapshot.Commit();
+            CollectionAssert.AreEqual(testValue, store.TryGet(testKey));
+
+            // Data saved to the leveldb from snapshot1 shall also be visible to snapshot2
```
Comment (cschuchardt88) [inline tests/Neo.Plugins.Storage.Tests/StoreTest.cs:190]: Can you check to see if `snapshot2` is visible to `snapshot1`

File: src/Plugins/LevelDBStore/Plugins/Storage/Snapshot.cs [hunk 2]
```diff
@@ -48,7 +48,8 @@ public void Dispose()
 
         public IEnumerable<(byte[] Key, byte[] Value)> Seek(byte[] prefix, SeekDirection direction = SeekDirection.Forward)
         {
-            return db.Seek(options, prefix, direction, (k, v) => (k, v));
+            return db.Seek(options, prefix, direction, (k, v) => (k, v))
```
Comment (shargon) [inline src/Plugins/LevelDBStore/Plugins/Storage/Snapshot.cs:51]: Could you explain what are you trying to fix?

Comment (cschuchardt88) [reply src/Plugins/LevelDBStore/Plugins/Storage/Snapshot.cs:51]: Im guessing, so the `snapshot` can read it items that are added.

**Comment by Jim8y** [comment]
@cschuchardt88 @shargon updated the tests and reverted the snapshot change, please review again.

File: src/Plugins/LevelDBStore/Plugins/Storage/Snapshot.cs (hunk 2 above)
Comment (Jim8y) [reply src/Plugins/LevelDBStore/Plugins/Storage/Snapshot.cs]: > Could you explain what are you trying to fix?

Reverted that change, cause in neo core, we can read added item in the snapshot, but here in the store the logic is different. Checked again, what we called snapshot in the core is actually snapshot cache.

**Review by cschuchardt88 — APPROVED** [review]

File: tests/Neo.Plugins.Storage.Tests/StoreTest.cs [hunk 3]
```diff
@@ -49,35 +58,143 @@ public void TestMemory()
         [TestMethod]
         public void TestLevelDb()
         {
-            using var plugin = new LevelDBStore();
-            TestPersistenceDelete(plugin.GetStore(path_leveldb));
+            ReSetStore();
```
Comment (shargon) [inline tests/Neo.Plugins.Storage.Tests/StoreTest.cs:61]: Already donde in Initialize, isn't it?

Comment (Jim8y) [reply tests/Neo.Plugins.Storage.Tests/StoreTest.cs:61]: Well, yes, but since there is an if check, should be fine.

**Review by shargon — APPROVED** [review]

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**